sql/src/main/java/io/crate/executor/transport/AlterTableOperation.java

```diff
@@ -191,6 +151,47 @@ public void fail(@Nonnull Throwable t) {
         return result;
     }
 
+    public CompletableFuture<Long> executeAlterTableRenameTable(AlterTableRenameAnalyzedStatement statement) {
+        DocTableInfo sourceTableInfo = statement.sourceTableInfo();
+        TableIdent sourceTableIdent = sourceTableInfo.ident();
+        TableIdent targetTableIdent = statement.targetTableIdent();
+
+        return closeTable(sourceTableInfo)
+            .thenCompose(r -> renameTable(sourceTableIdent, targetTableIdent))
+                .whenComplete((r, f) -> {
+                    TableIdent tableToOpen = targetTableIdent;
```

Comment:
You should track the state of a table here, and only re-open the table if the state was already open to begin with. If you are renaming a table that is _already_ closed, this operation would re-open the table, which is undesirable.